modules/help_editor/php/edit_help_content.class.inc

```diff
@@ -25,124 +27,133 @@ namespace LORIS\help_editor;
-        $this->tpl_data['subsection'] = $safeSubsection;
-        $this->tpl_data['helpID']     = $helpID ? $helpID : '';
-
-        if (!empty($helpID)) {
-            $help_file = HelpFile::factory($helpID);
-            $data      = $help_file->toArray();
-            $defaults['title']   = $data['topic'];
-            $defaults['content'] = trim($data['content']);
-            if (isset($_REQUEST['helpID'])) {
-                $this->tpl_data['module_name'] = "Help Editor";
-            } else {
-                $this->tpl_data['module_name'] = $defaults['title'];
-            }
 
+        if ($data['helpid']) {
+            $help_data       = HelpFile::factory($data['helpid'])->toArray();
+            $data['title']   = $help_data['topic'];
+            $data['content'] = htmlspecialchars_decode(trim($help_data['content']));
+
+            if (!isset($gets['helpID'])) {
+                $data['module'] = $help_data['topic'];
+            }
         }
-        // let escaped html display as intended in form
-        $defaults['content'] = htmlspecialchars_decode($defaults['content'] ?? '');
+
         // case where no help content exists
-        if (empty($defaults['title'])) {
-            if (!empty($_REQUEST['section'])) {
-                $defaults['title'] = str_replace("_", " ", $safeSection);
+        if (empty($data['title'])) {
+            if (!empty($data['section'])) {
+                $data['title'] = str_replace("_", " ", $data['section']);
             }
+
             // if not module and not instrument, should be a subtest
-            if (!empty($_REQUEST['subsection'])
-                && $_REQUEST['subsection'] != 'undefined'
+            if (!empty($data['subsection'])
+                && $data['subsection'] != 'undefined'
             ) {
-                $defaults['title'] = str_replace("_", " ", $safeSubsection);
+                $data['title'] = str_replace("_", " ", $data['subsection']);
             }
-            $this->tpl_data['module_name'] = $defaults['title'];
-        }
-        if (empty($defaults['content'])) {
-            $defaults['content'] = 'Under Construction';
-        }
-
-        if (isset($_SERVER['HTTP_REFERER']) ) {
-            $this->tpl_data['url'] = $_SERVER['HTTP_REFERER'];
         }
-        return $defaults;
-    }
-
-    /**
-     * Does the setup required for this page. By default, sets up elements
-     * that are common to every type of page. May be overridden by a specific
-     * page or specific page type.
-     *
-     * @return void
-     */
-    function setup()
-    {
-        parent::setup();
 
-        $this->addBasicText('title', 'Title');
-        $this->addBasicTextArea(
-            'content',
-            'Content',
-            [
-                'cols' => 140,
-                'rows' => 30,
-            ]
+        return (new \LORIS\Http\Response())->withBody(
+            new \LORIS\Http\StringStream(
+                '<div id="app" ' .
```

Comment:
Why is the help page returning a div with a series of data- attributes? It seems like it would be cleaner to return json.